### ISSUE TITLE
Poor Markdown Generation for Particular PDF

### ISSUE THREAD

**Issue by marty-sullivan** [pymupdf/RAG#75]
Here is a PDF file I'm testing with that produces some poor markdown (output is missing text that can be extracted by `pymupdf`). I believe this file was perhaps exported by Powerpoint into PDF.

[test.pdf](https://github.com/user-attachments/files/16229798/Sebastian_Raschka_TechTalk_Slides.pdf)

Using `pymupdf` by itself, the text generated is fine:

```python
import pymupdf

doc = pymupdf.open('test.pdf')
text = '\n'.join([page.get_text() for page in doc])
```

All text is accounted for in the output: 

[out.txt](https://github.com/user-attachments/files/16229823/out.txt)

Using `pymupdf4llm`:

```python
import pymupdf4llm

md = pymupdf4llm.to_markdown('test.pdf')
```

Much of the text is missing in the output, seems like titles are there but much of the slide content is missing:

[out.md](https://github.com/user-attachments/files/16229830/out.md)

Expected Behavior:

I would expect that, even if formatting cannot be accounted for, that any text in the pdf should still be included in the markdown.

**Comment by hewliyang** [comment]
try `.to_markdown('test.pdf', write_images=True)`, you will see that much content, especially the vector graphics are detected and outputted as images instead.

**Comment by JorjMcKie** [comment]
@hewliyang - thanks for your comment! You are quite right.

There are limits as to what the algorithm is capable to tell apart. It tries to detect (and separate from each other) images, drawings and text.
@marty-sullivan - your example contains many text surrounded by rectangular drawings with large rounded corners: we are unable to tell this apart from a significant vector graphic. So they are **_ignored as text_** but made available as images.

Do not forget: this is not an AI!

**Comment by marty-sullivan** [comment]
@hewliyang While this may work for some, it is not going to be compatible with the fundamentals of what I understand this module is supposed to do when text is available.

@JorjMcKie While I understand this project is early in development, I was very excited by how well it worked for most PDFs and other documents. And, while I'm sure there are fundamental complications to what I am bringing up, I *am not* asking for an AI, so I see that as an odd interpretation of what I outlined as my expected behavior.

Given that the underlying `pymupdf` is able to trivially extract text from these elements, I would expect the same behavior from this module, even if it simply attempts to extract text and set it in a MD quoteblock (perhaps detecting binary vs unicode as part of the decision as to whether to do that).

I hope I don't come off as too critical here, but if the underlying module can trivially extract text from vector graphics, and this module cannot and your viewpoint is "we are not going to fix that" -> Well, that essentially makes this module completely unviable for any real use.

**Comment by JorjMcKie** [comment]
I understand - don't take my comments too far.
What I am saying is this:
We cannot and will not do a full-blown analysis of vector graphics in this module, that safely differentiates between graphics made for cosmetic / outlining purposes (as they occur in your example) and "significant" vector graphics like Gantt Charts, bar charts, and the like.

What I **_am_** doing currently is determining whether the inner area (3 points away from the rectangle's border) only consists of fill color (or is empty). Your boxes have rounded corners which are not contained in stripes of at most 3 points breadth, so I regard them as "significant" vector graphics ... causing the text inside to be ignored.
That's the situation at hand.

If you have a suggestion how to handle this, you are most welcome.

**Comment by JorjMcKie** [comment]
@marty-sullivan thanks for your constructive ideas!
Maybe a combination of an additional parameter (dump_graphics ...) plus a somewhat improved significance detection is the way to go.

Currently looking at your rectangles with rounded corners: They cause the problem - normal rectangles wouldn't.
If I find a way to compute the enclosed area and compare it with the area of the "inner rectangle" may yield an improved significance check.
The rounded rectangle's area can be approximated when regarding all its defining points as a polygon, the area of which should be computable via Gauß's shoelace algorithm. If that result is close to the graphics rectangle, then we know that the object is insignificant.
I hope I am clear ...

**Comment by JorjMcKie** [comment]
Developed a new "significance check" for vector graphics that works as announced (shoelace algorithm).
Graphics consisting of "text decoration" only (highlights, rectangles with rounded corner and the like) are now identified as irrelevant / insignificant.
This however does not solve the general case as can be seen in this example:
![image](https://github.com/user-attachments/assets/52fdbb05-91bb-4ae2-a843-363dbc5104c3)

Green borders surround _**significant**_ vector graphics because of whatever complexity criteria. Red borders indicate unimportant / decoration-only drawings.

The algorithm has no way to detect that all graphics are on the same semantic level and should either be either regarded as vector graphic or forget all vector graphics and treat the text as text.

So in wrapping up, I think there is no way out of this situation other than introducing a new parameter which instructs to extract all text - whether or not contained inside a graphics (or image!) rectangle.

**Comment by JorjMcKie** [comment]
Fixed with v0.0.10.
